Ikuyadeu/vscode-R
# Improve development workflow

This PR makes working on the extension a bit easier.

By default, I disabled webpack and replaced the prelaunch task with a tsc watch task, which is considerably faster (especially on subsequent launches) than the webpack task. The script "scripts/enableWebpack.js" enables webPack again and can be called e.g. before publishing the extension. I also added some tsc watch tasks to compile the help/httpgd related typescript, but did not include these in the normal watch task, since most of the time they won't be changed anyways. Other possible modifications include

- [x] Adjusting the release actions to keep using webpack
- [x] Using yarn instead of npm. This basically comes down to replacing the lock file. In summary, yarn is compatible with the current package.json, but significantly faster. It's a bit less common than npm, apart from that I'm not aware of any drawbacks. For me personally, the switch would be worth it, but I also see the point that it might make running the extension from code a bit harder for new contributors.
- [x] Using subdirectories/other names for the typescript source files. Currently, there are some files (e.g. the help related stuff, or the upcoming liveshare files) that might as well be in their own folder. And I guess in most cases, we could remove the `r` from the beginning of file names, since they're all related to R anyways.

For the future, a slightly bigger change would be to set `"strictNullChecks": true` in tsconfig.json. This requires developers to work more cautiously when dealing with possibly undefined variables and requires rather many adjustments to the code base, since it cannot be turned off on a per-file basis (as opposed to eslint rules). But considering the rather strict rules implemented in `.eslintrc.yml`, I think it would make sense to also be strict here. 

**How to test:**
Behavior of the installed extension shouldn't change.
When running the extension from code, startup should be significantly faster than with webpack.
When the package is published on Github (as build artifact, pre-release, release) or the vscode marketplace, webpack should be enabled again.
